src/treelearner/cuda/cuda_single_gpu_tree_learner.hpp

```diff
@@ -118,6 +118,15 @@ class CUDASingleGPUTreeLearner: public SerialTreeLearner {
   score_t* cuda_gradients_;
   /*! \brief hessians on CUDA */
   score_t* cuda_hessians_;
+  /*! \brief whether boosting is done on CUDA */
+  const bool boosting_on_cuda_;
+
+  #ifdef DEBUG
+  /*! \brief gradients on CPU */
+  std::vector<score_t> host_gradients_;
+  /*! \brief hessians on CPU */
+  std::vector<score_t> host_hessians_;
+  #endif  // DEBUG
```

Comment:
are these codes used?

Reply:
Yes. These are used in DEBUG mode to check that the split of data and the split of histogram is consistent. See the `CheckSplitValid` method in cuda_single_gpu_tree_learner.cpp.